src/nccl_ofi_scheduler.c

```diff
@@ -53,6 +53,50 @@ void nccl_net_ofi_set_multiplexing_schedule(size_t size, int num_rails,
 	}
 }
 
+void nccl_net_ofi_set_multiplexed_round_robin_schedule(nccl_net_ofi_threshold_scheduler_t *scheduler,
+												   size_t size,
+												   int num_rails,
+					    						   size_t align,
+					    						   nccl_net_ofi_schedule_t *schedule)
+{
+	int rail_id;
+	int rail_group_size = 2;
+	nccl_net_ofi_mutex_lock(&scheduler->rr_lock);
+
+	/* Retieve and increment multiplex-round-robin counter; wrap around if required */
+	rail_id = scheduler->mux_rr_counter;
+	scheduler->mux_rr_counter += rail_group_size;
+	scheduler->mux_rr_counter = scheduler->mux_rr_counter == num_rails ? 0 : scheduler->mux_rr_counter;
```

Comment:
is there a reason why you don't use the modulo operation here? Like
```
scheduler->mux_rr_counter = (scheduler->mux_rr_counter + rail_group_size) % num_rails;
```
I think it is clearer like this, but it's just my opinion. This also has the added benefit of supporting cases where `num_rails` is not a multiple of `rail_group_size` (this is not the case today, but it doesn't harm to support that case).

Reply:
Thanks for the feedback. I have addressed this now.